src/screens/MoreScreen.js

```diff
@@ -1,33 +1,95 @@
+import {AppLoading} from 'expo';
 import React, {Component} from 'react';
-import {Text} from 'react-native';
-import {Icon} from 'react-native-elements';
+import {Platform, Text, View} from 'react-native';
+import {connect} from 'react-redux';
+import {Icon, Button, Card} from 'react-native-elements';
 
-import {CardSection, Card} from '../components/common/index';
+import {logUserOutOfFacebook} from '../actions';
+
+import {more_screen} from '../locale.en';
+import {navKeys} from '../constants';
 
 class MoreScreen extends Component {
+    componentDidMount() {
+        const {token, navigation: {navigate, state: {key}}} = this.props;
+
+        if (!token && key !== navKeys.LOGIN) {
+            navigate(navKeys.LOGIN);
+        }
+
+        this.props.navigation.setParams({onLogout: this.onLogout});
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const {token, navigation: {navigate, state: {key}}} = nextProps;
+
+        if (!token) {
+            navigate(navKeys.LOGIN);
+        }
```

Comment:
Same pattern as above. If we have no token, redirect to `LOGIN`. This is what will be the first thing to hear about the `logOut` action.